packages/cli-hydrogen/src/cli/services/deploy/upload.ts

```diff
@@ -0,0 +1,79 @@
+import {ReqDeployConfig, UploadDeploymentResponse} from './types.js'
+import {
+  CreateDeploymentResponse,
+  CreateDeploymentQuerySchema,
+  CreateDeploymentQuery,
+} from './graphql/create_deployment.js'
+import {WebPageNotAvailable} from './error.js'
+import {api, http, file} from '@shopify/cli-kit'
+import {zip} from '@shopify/cli-kit/node/archiver'
+
+export const createDeployment = async (config: ReqDeployConfig): Promise<CreateDeploymentResponse> => {
+  const headers = await api.buildHeaders(config.deploymentToken)
+  const client = await http.graphqlClient({
+    headers,
+    service: 'dms',
+    url: getDmsAddress(config.dmsAddress),
+  })
+
+  const variables = {
+    input: {
+      branch: config.commitRef,
+      commitHash: config.commitSha,
+      commitAuthor: config.commitAuthor,
+      commitMessage: config.commitMessage,
+      commitTimestamp: config.timestamp,
+    },
+  }
+
+  const response: CreateDeploymentQuerySchema = await client.request(CreateDeploymentQuery, variables)
+  return response.createDeployment
+}
+
+export const uploadDeployment = async (config: ReqDeployConfig, deploymentID: string): Promise<string> => {
+  let headers = await api.buildHeaders(config.deploymentToken)
+
+  // note: may need validation for invalid deploymentID? oxygenctl does it
+  // note: we may want to remove the zip that we create in in this step
+  const distPath = `${config.path}/dist`
+  const distZipPath = `${distPath}/dist.zip`
```

Comment:
Going to be moving this to be created in a temporary directory so that we do not pollute the `dist` directory that is created by Vite.

Reply:
This will be part of another PR later down the road.